_posts/2020-02-24-transit-gateway.md

```diff
@@ -0,0 +1,265 @@
+
+#### Transit Gateway Attachments
+For the VPC Workload accounts, when we create a Transit Gateway VPC Attachment which reference's the shared Transit Gateway resource, The VPC Attachment will automatically appear in the list of TGW Attachments in the Networking account.  Attachments can be generally thought of the network identifier for a network location, VPC's or VPN in our case. They can either be Added into TGW Route Tables as Associations or Propagations.
+
+#### Route Table Associations
+Transit Gateway Route Table Associations is where network traffic is initiated from(Source), but we don't use an IP Address CIDR block to define the source location, it is simply the name of the VPC or VPN Attachment. For example; if we want to create a TGW Route Table entry to route traffic from the on-premise network to a VPC in AWS, we know that the traffic would be initiated from the on-premise network 192.168.1.0/24, which AWS only knows about through the VPN Attachment, so the TGW Route Table entry would have the VPN Attachment as the Association and the VPC as the Propagation.  The source or initiator of traffic can only be associated to 1 Transit Gateway Route Table.  We will have 3 TGW Route tables in our deployment.
+
+#### Route Table Propagation
+Propagation is the path or next hop your traffic needs to take to arrive at its destination.  Once we add the VPC or VPN attachment to the TGW Route Table as a Propagation, the CIDR Block of the destination VPC Attachment will be automatically added to the Routes column as a propagated route, which you can override with a static IP CIDR Block destination, but you normally would not do this for VPC destinations, although you need to do this for VPN Propagations outside of the AWS network.  For example, if we are wanting to get to the on-premise network 192.168.1.0/24 from an AWS VPC, we would have the VPC as the Association (where traffic is initiated from) and VPN as the Propagation (The attachment that knows how to get to 192.168.1.0/24), followed by adding a static route for 192.168.1.0/24, which is linked to the VPN TGW Attachment Id and Viola! Jeff's your Uncle. 
+
+There is also the Black Hole option when it comes to adding Static Routes to the Transit Gateway Route Table.  In our architecture, because we want to isolate Workload VPC's from each other, we add TGW static routes for inter Workload-VPC traffic with Black-Hole as a destination.  
+
+#### Transit Gateway Route Table Cloudformation
+Below are Cloudformation snippets on how to deploy Transit Gateway Route tables as per the architecture diagram we have illustrated earlier on. We will need to deploy the 3 Transit Gateway Route Tables, Route Table Associations, Route Table Propagations and Routes. The naming will follow what we have in the architecture diagram. 
+
+    VpnToVpcRouteTable:
```

Comment:
are these in code snippets?

Reply:
Yep, they appear as code snippets.  Did you want to have the contacts email shown twice? at the end of the sentence and at the footer of the page, which is part of the template..  or shall I use the "lets get started link" https://www.mechanicalrock.io/lets-get-started/